service/src/main/java/com/ge/predix/acs/privilege/management/dao/GraphGenericRepository.java

```diff
@@ -402,7 +401,7 @@ private void searchAttributesWithScopes(final E entity, final Vertex vertex, fin
                         attributes.addAll(deserializedAttributes);
                         // This enforces the limit on the count of attributes returned from the traversal, instead of
                         // number of vertices traversed. To do the latter will require traversing the graph twice.
-                        checkTraversalLimitOrFail(attributes);
+                        checkTraversalLimitOrFail(entity, attributes);
```

Comment:
To add to the previous comment, see [here](https://docs.oracle.com/javase/8/docs/api/java/util/stream/package-summary.html) for a discussion of "intermediate" vs "terminal" operations in Java 8 streams. The bottom line is that streams are lazy. Nothing happens until you call a terminal operations such as `forEach()`, `collect()`, `reduce()`, etc.